bin/dwh-migration-dumper

```diff
@@ -7,4 +7,7 @@ if [ ! -x "$BIN" ] ; then
   (cd "${BASE_DIR}" && ./gradlew --parallel :dumper:app:installDist)
 fi
 
-exec "$BIN" "$@"
+# "ALL-UNNAMED" option is required as a workaround for Snowflake JDBC Driver error.
+_JAVA_OPTIONS="--add-opens=java.base/java.nio=ALL-UNNAMED"
```

Comment:
1. What if the `_JAVA_OPTIONS` variable already contains some value? Would it make sense to preserve it `_JAVA_OPTIONS="$_JAVA_OPTIONS --add-opens..."`?

2. It seems from the linked manual that the parameter depends on the Java version. This workaround is specific for JDK 17 and 18, while for JDK 16 another workaround is suggested: `-Djdk.module.illegalAccess=permit`.

    The `_JAVA_OPTIONS` variable seems to be specific to a particular JVM implementation. The `JAVA_TOOL_OPTIONS` variable is suggested as a more general solution: https://stackoverflow.com/q/28327620

    Another possibility is to use `JAVA_OPTS` or `DWH_MIGRATION_DUMPER_OPTS`, although I'm not sure if they would work the same as `_JAVA_OPTIONS` or `JAVA_TOOL_OPTIONS`: https://github.com/gradle/gradle/blob/bda658e496d0f001bbad8c59d7cd8b6f0ad85067/platforms/jvm/plugins-application/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt#L244

Reply:
1. Nice idea. Change applied.
2.a. I tried using Java 16 with this fix, and it still works.
2.b. I think it is desirable that _JAVA_OPTIONS "wins over" command-line arguments and _JAVA_TOOL_OPTIONS, because we do not want other changes to accidentally overwrite this break-preventing option. Also, it is recommended by the official Arrow site: https://arrow.apache.org/docs/java/install.html#java-compatibility